core/src/main/java/hudson/model/Executor.java

```diff
@@ -220,6 +220,13 @@ public void run() {
             try {
                 workUnit.context.synchronizeStart();
 
+                // this code handles the behavior of null Executables returned
+                // by tasks. In such case Jenkins starts the workUnit in order 
+                // to report results to console outputs. 
+                if (executable == null) {
+                    throw new Error("The null Executable has been created for "+workUnit+". The task cannot be executed");
```

Comment:
This is not very helpful for two reasons.

First, an `Error` implies that something is seriously broken, which is not necessarily true. It may simply be that a project was scheduled, and then just before it was passed off from the queue to an `Executor` it got disabled.

Second, the message does not specify what the `task` was that cannot be executed!